text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,505 @@
+This requires a 1:1 mapping of parameters and tags that need to be passed from the stack to the stack set. 
+Customer can deploy to existing stack sets, but requires some migration steps for importing an existing stack set into the resource `AWS::CloudFormation::StackSet` before before being able to use `cdk.CFNStackSet`.
+
+### Why should we _not_ do this?
+
+We might not want to do this since CDK does provide the `cdk.CFNStackSet` construct, which could accomplish similar behavior. 
+The intermdiate stack may not be ideal for some customers, and may require large migration effort for customers who manage many stacksets.
+
+### What changes are required to enable this change?
+
+To enable this feature, we would need to introduce a new StackSet construct for customers to extend, a new assembly/artifact type for the Cloud Assembly, stack set synthesizer, and deployment code that creates/updates stack sets using AWS CloudFormation APIs.
+
+### Is this a breaking change?
+
+No.
+
+### What are the drawbacks of this solution?
+
+The handling of file assets is more complex than a normal stack CDK app since stack set instances are deployed independently from CDK context and exist in different accounts and regions. 
+This is dicussed more in the design details.
+
+### What alternative solutions did you consider?
+
+An alternative is the usage of `cdk.CFNStackSet`, which is not ideal for customers with existing stack sets since there is significant migration effort.
+
+### What is the high level implementation plan?
+* Create StackSet core construct
+* Create StackSet artifact type in the Cloud Assembly
+* Create Bootstrapless StackSet Synthesizer
+* Update CLI to collect StackSet artifact types in addition to Stacks
+
+### Are there any open issues that need to be addressed later?
+
+* CDK asset management
+* Future support for service managed stack sets
+* CDK Pipelines
+
+---
+# Implemenation Details
+
+# Requirements
+
+## In Scope Features
+
+* Create Self-Managed Stack Set
+  * This should create the stack set directly, as opposed to the [CfnStackSet](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-cloudformation.CfnStackSet.html) construct which creates a CloudFormation stack that deploys a stack set
+* Update Self-Managed Stack Set
+  * Update includes updating **all existing** stack set instances 
+  * Set Operation Preference Configuration
+
+## Out of Scope Features
+
+* Delete Self-Managed Stack Set
+* Service Managed Stack Set
+* Add/Remove stack set instances through CDK
+* Update to only specific stack set instances
+* Stack Set Instance Level Parameter Overrides
+* CDK Assets (CDK File Assets/CDK Docker Assets)
+  * More information on CDK asset management and complexities described in the [File Assets Management](#4-file-assets-management) section.
+
+## Assumptions
+
+* Management Account contains Stack Set administration role and child accounts contains Stack Set execution role for enabling stack set operations
```

Comment:
Can we settle on the terminology "parent account" and "child account" here? " Management Account" just sounds weird (and is also weirdly capitalized).

Reply:
Fixed in all occurences